Juriba.Dashworks/Public/New-CustomField.ps1

```diff
@@ -0,0 +1,119 @@
+function New-CustomField {
+    <#
+    .SYNOPSIS
+
+    Creates a new custom field.
+
+    .DESCRIPTION
+
+    Creates a new custom field using the Dashworks API v1.
+
+    .PARAMETER DashworksSession
+
+    Hash table containing Domain, Port and API Key for your Dashworks instance.
+
+    .PARAMETER Name
+
+    Name of the new custom field.
+
+    .PARAMETER CSVColumnHeader
+
+    CSV Column Header for the new custom field. Restricted to alphanumeric characters.
+
+    .PARAMETER Type
+
+    Type of the new custom field. One of Text, MultiText, LargeText, Number or Date.
+
+    .PARAMETER ObjectTypes
+
+    Object types that this new custom field applies to. Accepts multiple selections. One or more of Device, User, Application, Mailbox.
+
+    .PARAMETER IsActive
+
+    Set the new custom field to active or inactive. Defaults to Active.
+
+    .INPUTS
+
+    None. You cannot pipe objects to Add-Extension
+
+    .OUTPUTS
+
+    None.
+
+    .EXAMPLE
+
+    PS> New-CustomField -ObjectTypes Device, User -Name "MyNewCustomField" -CSVColumnHeader "mynewcustomdfield" -Type Text -IsActive $true -DashworksSession $DashworksSession
+
+    #>
+
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute(
+        "PSUseShouldProcessForStateChangingFunctions",
+        "",
+        Justification = "API endpoint does not support ShouldProcess."
+    )]
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]
+        [System.Collections.Hashtable]$DashworksSession,
```

Comment:
The hashtable input make it harder for someone to consume as they have to look at help, and means we cant use the parameter validation helpers? If someone wants to avoid specifying common parameters everytime they could use splatting.

Reply:
Agreed. I'll expand it into the three individual parameters. $Instance, $Port, $APIKey